### PR TITLE
UIStatusBarStyle now changes correctly when a device is in dark mode

### LINKED ISSUE #8294: [Bug] UIStatusBarStyle does not appear to change when device is in dark mode
### Description

If an iOS device with iOS 13 is in dark mode then it is not possible to change the status bar color. Even if you [opt-out of dark mode](https://montemagno.com/opt-out-of-dark-mode-on-ios-android-uwp-and-macos/) (via @jamesmontemagno) the status bar will still be white while the device is in dark mode.

Improving on #4343 could also possibly help this. 

### Steps to Reproduce
1. Add following to your Info.plist, 
```
<key>UIViewControllerBasedStatusBarAppearance</key>
<false/>
```
2. In a Xamarin.Forms project create a NavigationPage with a BarBackgroundColor of white, BarTextColor of black.
3. Change device to dark mode.
4. Run and see that you can't see the status bar.
Bonus - 5. Change device to light mode, things will work as expected.

### Expected Behavior
Stats bar is black to be visible on the white BarBackgroundColor.

### Actual Behavior
Status bar is always white, even if you opt-out of dark mode with
```
<key>UIUserInterfaceStyle</key>
<string>Light</string>
```
in your Info.plist, making it hard to see on the white BarBackgroundColor.

### Basic Information
- Version with issue: Forms 4.3.0.947036
- Last known good version: N/A
- Platform Target Frameworks: 
  - iOS:  13.0+ (tested on iPhone, unsure if this effects iPad and iPod)

### Screenshots

statusbar_issue

### Reproduction Link

[StatusBarTest.zip](https://github.com/xamarin/Xamarin.Forms/files/3787128/StatusBarTest.zip)

### PR DESCRIPTION
**Pull request by beeradmoore**
### Description of Change ###
With the introduction of iOS 13, dark mode and the new UIStatusBarStyle.DarkContent enum status bar color set automatically from a NavigationPage has been incorrect. When the device is in dark mode and tries to update the StatusBarStyle to UIStatusBarStyle.Default this property behaves differently depending on if the device is in dark/light mode (or if the app is forced into light/dark mode with the UIUserInterfaceStyle key in Info.plist) and we get an unexpected status bar style.

Instead now when running on iOS13 it will use the new UIStatusBarStyle.DarkContent enum instead of UIStatusBarStyle.Default to display the expected UIStatusBarStyle.

### Issues Resolved ### 
- fixes #8294

### API Changes ### 
None

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
On iOS 13 when a device is using dark mode the correct UIStatusBarStyle is now set. 

### Before/After Screenshots ### 

beforeafter_small

### Testing Procedure ###
Issue8294 was added to Xamarin.Forms.Controls.Issues.
To test, change UIViewControllerBasedStatusBarAppearance to false in the Xamarin.Forms.ControlGallery.iOS Info.plist. This allows us to not rely on each ViewController to specify its UIStatusBarStyle and instead we can set it with `UIApplication.SharedApplication.StatusBarStyle = ...;`

On an iOS device not in dark mode launch the control gallery and open issue 8294. 
Tapping light and dark you should see the status bar text color change correctly.
Now close the control gallery app, put the device in dark mode, launch the control gallery app and repeat the above steps. 

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs [hunk 1]
```diff
@@ -679,7 +679,14 @@ void SetStatusBarStyle()
 			if (statusBarColorMode == StatusBarTextColorMode.DoNotAdjust || barTextColor.Luminosity <= 0.5)
 			{
 				// Use dark text color for status bar
-				UIApplication.SharedApplication.StatusBarStyle = UIStatusBarStyle.Default;
+				if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
```
Comment (jfversluis) [inline Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs]:
```suggestion
				if (Forms.IsiOS13OrNewer)
```

**Comment by beeradmoore** [comment]
@jfversluis I didn't see the fix for the conflict in your code suggestion above but I did some PRs and fixed that.

**Comment by jfversluis** [comment]
The suggestion was just for the piece of code to check the iOS version. Typically you could do a rebase to resolve any conflicts, but this works too! 

Looks good to me know. Only hurdle we still have to take is that we don't build against Xcode 11 in our pipeline yet :(

**Comment by IeuanWalker** [comment]
@beeradmoore @samhouts @jfversluis @jsuarezruiz 
Hi, I think this might have caused an issue for me.

Updating from 4.4 to 4.5 and the only thing in the release notes related to the status bar is this fix.

In my project I set the Status bar background colour, but this is now getting ignored after updating to 4.5.

```csharp
// style ios navbar
UIColor barTintColor = UIColor.FromRGB(5, 60, 61);
UIColor tintColor = UIColor.White;
UIApplication.SharedApplication.SetStatusBarStyle(UIStatusBarStyle.DarkContent, true);
UINavigationBar.Appearance.BarTintColor = barTintColor;
UINavigationBar.Appearance.TintColor = tintColor;
UINavigationBar.Appearance.TitleTextAttributes = new UIStringAttributes() { ForegroundColor = UIColor.White };
UINavigationBar.Appearance.Translucent = true;
```

**Comment by beeradmoore** [comment]
Hey @IeuanWalker , I tried with my a modified sample from #8294 to replicate your issue (
[StatusBarTest-status-color.zip](https://github.com/xamarin/Xamarin.Forms/files/4269169/StatusBarTest-status-color.zip)) but was not able to make it change from either AppDelegate.cs or from within a button on those pages. Sample points to XF 4.3 but I also tried on 4.4.

Could you please file a new bug report and reference this PR?

**Comment by IeuanWalker** [comment]
@beeradmoore done

**Pull request merged**